middleware/normalizeParentIds.js

```diff
@@ -32,7 +32,9 @@ function normalizeParentIds(place) {
   if (place) {
     placeTypes.forEach(function (placeType) {
       if (place[placeType] && place[placeType].length > 0 && place[placeType][0]) {
-        let source = 'whosonfirst';
+        // This is a solution for geonames hack.
+        // We can store in ES the source and defaulted to wof for backward compatibility.
+        let source = _.get(place, `${placeType}_source`, ['whosonfirst']);
```

Comment:
```javascript
_.head('foo');
"f"
```

😖

Reply:
In fact, `xxx_source` should be an array [pelias/model/Document.js#L393-L396](https://github.com/pelias/model/blob/8aef69949e3808d8f9f16865b8b477d9ec802f5f/Document.js#L393-L396)

I wanted to be consistent with what there was before and the [message in the addParent function](https://github.com/pelias/model/blob/8aef69949e3808d8f9f16865b8b477d9ec802f5f/Document.js#L346-L367).

```
    note: the rationale for setting this field as 'null' instead of 'undefined'
    is so that each of the fields 'line-up'.
    == if you add a parent property with no abbreviation, as such:
    addParent( 'region', 'foobar', '1' )
    doc:
      parent.region       = [ 'foobar' ]
      parent.region_id    = [ '1' ]
      parent.region_a     = [ null ]
    == and then you add another parent property such as:
    addParent( 'region', 'bingobango', '2', 'bingo' )
    doc:
      parent.region       = [ 'foobar', 'bingobango' ]
      parent.region_id    = [ '1',      '2' ]
      parent.region_a     = [ null,     'bingo' ]
    == you can now be sure that the abbreviation 'bingo' belongs to '2' and not '1'.
```

***

In fact, this message is not really correct because there is a deduplication on each field.. But it's another issue and this feature is not yet used is pelias :sweat_smile: 

```js
// Code is better than words
doc.addParent('region', 'Germany', '1', 'DE', 'whosonfirst')
doc.addParent('region', 'Germany', 'rel/1', 'DE', 'openstreetmap')
assert(doc.parent, {
  "region": ["Germany"], // This should be ["Germany", "Germany"]
  "region_id": ["1", "rel/1"],
  "region_a": ["DE"], // This should be ["DE", "DE"]
  "region_source": ["whosonfirst", "openstreetmap"]
}
```

Reply:
Oh yeah good point about the deduplication, I think my code comment pre-dated that work.
But yeah, probably better as it is, because the 'field length' affects the scoring, so dupes throw the scoring off.

Makes total sense, I think I just got burned in the past by ES returning Array/String depending on how it was inserted, so I'm a bit paranoid.

I *think* in this case it should be ok, with the only exception being someone creating an index using something other than `pelias/model` which I think is unlikely.

:+1: good to merge.